local-interchain/interchain/config.go

```diff
@@ -45,14 +33,43 @@ func LoadConfig(installDir, chainCfgFile string) (*types.Config, error) {
 	chainsDir := filepath.Join(installDir, "chains")
 	cfgFilePath := filepath.Join(chainsDir, configFile)
 
-	config, err := loadConfig(config, cfgFilePath)
+	bytes, err := os.ReadFile(cfgFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(bytes, &config)
 	if err != nil {
 		return nil, err
 	}
 
 	log.Println("Using directory:", installDir)
 	log.Println("Using chain config:", cfgFilePath)
 
+	return setConfigDefaults(&config), nil
+}
+
+func LoadConfigFromURL(url string) (*types.Config, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		log.Fatalln(err)
+	}
```

Comment:
It is the callers responsibility to ensure the response body of an http request is closed after use, failure to do so can lead to resource leak, etc.

```suggestion
	resp, err := http.Get(url)
	if err != nil {
		log.Fatalln(err)
	}
	defer resp.Body.Close()
```

Reply:
great catch